src/routes/+page.svelte

```diff
@@ -0,0 +1,80 @@
+<script lang="ts">
+    import { onMount } from 'svelte';
+    import Gallery from '../components/Gallery.svelte';
+    import MovieDetail from '../components/MovieDetail.svelte';
+    import NavBar from '../components/NavBar.svelte';
+    import '/src/global.css';
+    import type { Movie } from '../interfaces/Movie';
+    import { page } from '$app/stores';
+    import { goto } from '$app/navigation';
+
+    let movies: Movie[] = [];
+    let currentIndex = -1;
+    let showDetail = false;
+    let currentMovie = movies[currentIndex];
+    let franchiseId: number;
+
+    $: {
+        const { movies: loadedMovies } = $page.data;
+        if (loadedMovies) {
+            movies = loadedMovies;
+        } else {
+            console.log("No movies loaded");
+        }
+    }
+
+    function nextMovie() {
+        setInterval(() => {
+            if (!showDetail) {
+                currentIndex = (currentIndex + 1) % movies.length;
+            }
+            currentMovie = movies[currentIndex];
+            showDetail = !showDetail;
+        }, 20000);
+    }
+
+    function backToGallery() {
+        showDetail = false;
+    }
+
+    function handleFranchiseSelected(event: any) {
```

Comment:
This event should probably be something like:

`event: CustomEvent<{ detail: InsertFranchiseTypeHere }>`

Reply:
I changed the event type this should be better right ? 

https://github.com/BowenGeense/MovieShowcase/pull/3/files#diff-1e1270da740e81dd13f8a65057582b0fce8a5fb3817ef913f3e90d7f82c4e9df